// openfsm-transaction-dto/TransactionDto.js
class TransactionDto {
  constructor(data = {}) {
    this.transactionId = data.transaction_id || undefined;
    this.accountId = data.account_id || undefined;
    this.transactionType = data.transaction_type || undefined;
    this.amount = data.amount || undefined;
    this.status = data.status || undefined;
    this.referenceId = data.reference_id || undefined;
    this.createdAt = data.created_at || undefined;
    this.updatedAt = data.updated_at || undefined;
  }

  // Getters
  getTransactionId() {
    return this.transactionId;
  }

  getAccountId() {
    return this.accountId;
  }

  getTransactionType() {
    return this.transactionType;
  }

  getAmount() {
    return this.amount;
  }

  getStatus() {
    return this.status;
  }

  getReferenceId() {
    return this.referenceId;
  }

  getCreatedAt() {
    return this.createdAt;
  }

  getUpdatedAt() {
    return this.updatedAt;
  }

  // Setters
  setTransactionId(transactionId) {
    this.transactionId = transactionId;
  }

  setAccountId(accountId) {
    this.accountId = accountId;
  }

  setTransactionType(transactionType) {
    this.transactionType = transactionType;
  }

  setAmount(amount) {
    this.amount = amount;
  }

  setStatus(status) {
    this.status = status;
  }

  setReferenceId(referenceId) {
    this.referenceId = referenceId;
  }

  setCreatedAt(createdAt) {
    this.createdAt = createdAt;
  }

  setUpdatedAt(updatedAt) {
    this.updatedAt = updatedAt;
  }
}

// Экспортируем класс для использования в других модулях
module.exports = TransactionDto;
